Allow Bob to accept a swap

thomaseizinger opened the pull request:
This PR adds two routes to the HTTP API:

- accept
- decline

These are the two actions that Bob can perform when presented with a swap request. At the moment, only `accept` is implemented but this is the much more difficult one. If he accepts, his response is used for two things:

- it is sent back to Alice
- his own state machine is advanced into the next state and starts listening for Alice to deploy the first contract

Commit:
Add routes for accept/decline action

thomaseizinger committed:
Store pending responses from bam_api

This introduces a `PendingResponses` component which is used by Bob's
SwapRequestHandler to store pending responses. The http_api will use
this component to send the user's response to the swap request which
will result in Bob's state machine advancing to the next state AND the
response will be sent to Alice.

thomaseizinger committed:
Move HttpApiProblemStdError into dedicated module

Since the type is now used in to different routes, it should really be
it's own module.

D4nte reviewed (changes requested):
Well done @thomaseizinger for finishing off this PR! Few minor comments

thomaseizinger committed:
Improve log message to reveal underlying error

thomaseizinger committed:
Remove intermediate variable

bonomat reviewed (commented):
Good job. I just added some comments.

thomaseizinger committed:
Avoid runtime errors

thomaseizinger committed:
Improve error message in case of failure

thomaseizinger committed:
Remove type from problem

Error types help clients to dynamically react to them. We don't need
that in this case.

Commit:
Correct error message
